### PR TITLE
Bug: Return environments for each connection in local provider

### LINKED ISSUE #11525: [Local Provider] Not able to assign connection to environment
### Current Behavior
On the connections page in Meshery, after logging in as a local provider, the environments dropdown shows a success message when I assign an environment, but the dropdown doesn't get updated.

### Expected Behavior
You can see the expected behavior by logging in as a remote provider. Ideally, after the success message, the environment checkbox should be selected in the dropdown, and on refresh, it should show the assigned environments.

### Screenshots/Logs

https://github.com/user-attachments/assets/108f953e-e0f8-41b3-a158-9258e8f8fa03

### My investigation
I found that this issue occurs because the local provider API does not return assigned environments with the connection GET request API, whereas the remote provider does. That's why, local provider API needs to be updated to return the environments as well.

### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by Aijeyomah**
**Notes for Reviewers**

This PR fixes #11525
-  Return environments for each connection for local provider `GetConnections` 

https://github.com/user-attachments/assets/52ca125f-223f-4281-a126-7803dfd90177

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/connection_persister.go [hunk 1]
```diff
@@ -49,11 +50,17 @@ func (cp *ConnectionPersister) GetConnections(search, order string, page, pageSi
 	count := int64(0)
 
 	connectionsFetched := []*connections.Connection{}
-
 	query.Table("connections").Count(&count)
-
+	environmentsFetched := []environments.EnvironmentData{}
```
Comment (Yashsharma1911) [inline server/models/connection_persister.go:54]: Please use from schemas package, checkout environment_persister.go file for implementation, and here is where environment is defined - https://github.com/meshery/schemas/blob/master/models/v1beta1/environment.go#L12
```suggestion
	environmentsFetched := []v1beta1.Environment{}
```

**Review by Yashsharma1911 — COMMENTED** [review]
Thankyou for raising PR after this update we should we good to go

File: server/models/connection_persister.go (hunk 1 above)
Comment (Yashsharma1911) [reply server/models/connection_persister.go:54]: Ping me when this is done

**Comment by Yashsharma1911** [comment]
UI is updating slow, hmm API latency causing issue 🤔

File: server/models/connection_persister.go (hunk 1 above)
Comment (Aijeyomah) [reply server/models/connection_persister.go:54]: > environmentsFetched := []v1beta1.Environment{}

This environment struct is not compatible with the connection.Environment struct which is `[]environments.EnvironmentData`

Comment (Yashsharma1911) [reply server/models/connection_persister.go:54]: I see a complete migration of connection needed to make sure we are using v1beta1, never mind you can leave this review thanks for confirming

**Comment by Aijeyomah** [comment]
> , hmm API latency causing issue 🤔

It's actually really fast. My laptop was hanging while recording the screen.

**Comment by leecalcote** [comment]
I wonder what's causing [this step](https://github.com/meshery/meshery/actions/runs/10258147072/job/28380543476?pr=11559#step:7:1) to run for 5 hours. Is this an item that you can bring this week's Meshery Build and Release meeting, @Aijeyomah?

**Review by shubham251972 — APPROVED** [review]
LGTM

**Comment by Aijeyomah** [comment]
> I wonder what's causing [this step](https://github.com/meshery/meshery/actions/runs/10258147072/job/28380543476?pr=11559#step:7:1) to run for 5 hours. Is this an item that you can bring this week's Meshery Build and Release meeting, @Aijeyomah?

Yeah, sure

**Review by Yashsharma1911 — APPROVED** [review]
LGTM thankyou

**Pull request merged**